### ISSUE TITLE
Roadmap Ahead

### ISSUE THREAD

**Issue by vincentqb** [pytorch/audio#1163]
In the last six months, torchaudio has seen exciting work released as part of two major releases and a new [audio command recognition tutorial](https://pytorch.org/tutorials/intermediate/speech_command_recognition_with_torchaudio.html).

* [torchaudio 0.7.0](https://github.com/pytorch/audio/releases/tag/v0.7.0) and [0.7.2](https://github.com/pytorch/audio/releases/tag/v0.7.2) expand its support for models and [end-to-end applications](https://github.com/pytorch/audio/tree/master/examples). It offers training example pipelines for Speech Recognition with Wav2Letter and LibriSpeech; for Text-to-Speech with WaveRNN as a vocoder and LibriTTS; added a vocoder based on the WaveRNN model; and Source Separation with ConvTasNet model and wsj0-mix.
* [torchaudio 0.6.0](https://github.com/pytorch/audio/releases/tag/v0.6.0) includes a new model module (with wav2letter included), new functionals (contrast, cvm, dcshift, overdrive, vad, phaser, flanger, biquad), datasets (GTZAN, CMU), and a new optional sox backend with support for torchscript. torchaudio now also supports Windows, with the soundfile backend.

There are many exciting work elements that are planned.

* We want to ensure that torchaudio works well for large scale training. To do so, we plan on adding support a large-scale training reference task using wav2vec on librivox. This would also allow us to provide pre-trained versions of the model to the community. In addition, given the ubiquity of transformers, we want to investigate what audio specific abstractions would be beneficial to expose within torchaudio.
* We want to build on the [speech recognition pipeline](https://github.com/pytorch/audio/tree/master/examples/pipeline_wav2letter) we added within torchaudio. In particular, we will investigate the addition of beam search, and a 4-gram language model, see [here](https://github.com/pytorch/audio/issues/913). Along those lines, we also plan to partner with GTN to provide an example using both GTN and torchaudio. We also want to add new transforms for in-memory codec encoding and decoding to be used as data augmentation, see [here](https://github.com/pytorch/audio/issues/1115), and to expose the Kaldi pitch feature directly within torchaudio, see [here](https://github.com/pytorch/audio/pull/1063). We also plan on adding the RNN transducer loss as a new loss function within torchaudio, see [here](https://github.com/pytorch/audio/pull/1137).
* Providing high-performance data loading and media decoding experience is important. We will add a loading function that performs a format-based backend dispatch mechanism with the choice of fast portable decoding library, see [here](https://github.com/pytorch/audio/issues/1000). Moreover, given that many applications work on streaming data, we want to provide new audio streaming abstractions within torchaudio, see [here](https://github.com/pytorch/audio/issues/1072).
* To make torchaudio available in environments where python may not be available, we also plan on building the C++ extension outside of Python as libtorchaudio.

The goal of torchaudio is to accelerate research through novel, production-ready building blocks. As such, we would love to hear feedback on the plan, so make sure to give feedback on the plan to us, @mthrok and @vincentqb!
